### ISSUE TITLE
FastApi Error : "RuntimeError: Response content longer than Content-Length"

### ISSUE THREAD

**Issue by sidraw-sys** [tiangolo/fastapi#4949]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#dummy list for posts
my_posts=[{"title":"Post 1","content":"Content of Post 1","id":1 },
          {"title":"Post 2","content":"Content of Post 2","id":2 },
          {"title":"Post 3","content":"Content of Post 3","id":3 }  
         ]

#function to get the index of the post which I am looking to delete by matching the ID
def find_index_post(id):
    for i, p in enumerate(my_posts):
        if p['id']==id:
            return i

#delete request created using FastApi
@app.delete("/posts/{id}",status_code=status.HTTP_204_NO_CONTENT)
def delete_post(id:int):
    index=find_index_post(id)
    if index==None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=f"Post with Id {id} does not 
        exist!")
    
    my_posts.pop(index)
    return Response(status_code=status.HTTP_404_NOT_FOUND)
```

### Description

I have started learning FastApi recently and have been following one youtube tutorial for lessons.

The issue i have been facing with this code is whenever I run delete_post request on Postman I get below error on uvicorn server: 
             "return Response(status_code=status.HTTP_204_NO_CONTENT)
              TypeError: Response.__init__() got an unexpected keyword argument 'status_code'" 
However, in the instructors video, this is working fine without any errors.

I searched and found few issues related to this on here which I tried to follow but couldn't get myself to resolve the error. I tried following few solutions as provided inside below ISSUE ID but to no avail-
          " Response content longer than Content-Length error for DELETE and NoContent #4939"

The two prominent errors I am getting are:
- "return Response(status_code=status.HTTP_204_NO_CONTENT)
   TypeError: Response.__init__() got an unexpected keyword argument 'status_code'
   ***This is the original code which i am following from the youtube lesson

- raise RuntimeError("Response content longer than Content-Length")
   RuntimeError: Response content longer than Content-Length
  ***This is the error i am getting when I try to follow the solution provided in Issue id "4939".
 

  
Kindly assist, Thanks!

### Operating System

Windows

### FastAPI Version

0.78.0

### Python Version

Python 3.10.4

### Additional Context

NA

**Comment by sidraw-sys** [comment]
Also , on trying the solution as told to follow in #4939, i am getting an error i have mentioned at bottom:

///Updated Code for delete request:

@app.delete("/posts/{id}",status_code=status.HTTP_204_NO_CONTENT,response_class=Response)
def delete_post(id:int):
    index=find_index_post(id)
    if index==None:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,detail=f"Post with Id {id} does not exist!")
    
    my_posts.pop(index)
    return None

///Getting below error:
response = actual_response_class(response_data, **response_args)
TypeError: Response.__init__() got an unexpected keyword argument 'background'

**Comment by sidraw-sys** [comment]
Hi Ben, 
Sorry, but can you explain the resolution in bit detail, what modifications can I make in order to resolve the issue.
I went through the pull request but being a newbie I was not able to comprehend the tests and handling you did in there.
Thanks for your help, kindly assist if possible in detail.

**Comment by florimondmanca** [comment]
I believe this is a duplicate of #4939.

**Comment by tiangolo** [comment]
Thanks @sidraw-sys! It seems this was handled in https://github.com/tiangolo/fastapi/pull/5145, available since FastAPI `0.79.0`.

If that solves it, you could close the issue. Thanks! :cake:

Thanks for the help @florimondmanca! :bow:

**Comment by gazpachoking** [comment]
Commenting on this ticket as well, since it's still open.

This seems to be an issue for me still on fastapi 0.82.0. It can be reproduced like this:
```python
from fastapi import FastAPI, HTTPException
import uvicorn

app = FastAPI()


@app.get("/")
async def root():
    raise HTTPException(204)

if __name__ == '__main__':
    uvicorn.run(app, debug=True)
```
Visiting the page results in:
```
RuntimeError: Response content longer than Content-Length
```

**Comment by estronnom** [comment]
Stuck with the same problem here

**Comment by JarroVGIT** [comment]
@gazpachoking and @estronnom what version of Uvicorn are you using?

**Comment by estronnom** [comment]
@JarroVGIT 0.18.3

**Comment by iudeen** [comment]
Yes, I can confirm this seems to be a bug.

**Comment by gazpachoking** [comment]
> @gazpachoking and @estronnom what version of Uvicorn are you using?

> @JarroVGIT 0.18.3

Same. 0.18.3

**Comment by iudeen** [comment]
It works fine with hypercorn. 

```
Request URL: http://127.0.0.1:8000/
Request Method: GET
Status Code: 204 
Remote Address: 127.0.0.1:8000
Referrer Policy: strict-origin-when-cross-origin
```

**Comment by iudeen** [comment]
I think I got the issue, will make a PR.

**Comment by iudeen** [comment]
The reason this one keeps escaping the tests IMO is because FastAPI returns the given status code even if starlette raised 500 - Internal Server Error

**Comment by tiangolo** [comment]
Thanks everyone! This should have been fixed by @iudeen in https://github.com/tiangolo/fastapi/pull/5365 :rocket: 

This is available in the next version of FastAPI released in the next hours (the current is `0.82.0`). :tada:
